icu4c/source/i18n/formatted_string_builder.cpp

```diff
@@ -260,6 +260,11 @@ int32_t FormattedStringBuilder::prepareForInsert(int32_t index, int32_t count, U
         fZero -= count;
         fLength += count;
         return fZero;
+    } else if (fZero + fLength + count < 0 || fLength + count < 0 || fZero + fLength < 0) {
```

Comment:
Suggestion: use `uprv_add32_overflow` and related methods from putilimp.h to check for integer overflow, rather than checking for negative values, which is not bulletproof